nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -2061,6 +2219,172 @@ mod tests {
         logctx.cleanup_successful();
     }
 
+    #[tokio::test]
```

Comment:
Great test!

I usually do not like to add test only code to production methods. However, in this case, I think ensuring that CRDB transaction serialization works as we expect warrants it. This also serves as a useful regression in case that behavior changes.

Reply:
I wonder if `select_for_update`, which should take a write lock on the target table, or at least target row, also blocks reading from that table. I'm not sure if that would be a problem, and don't think it warrants a test. I'm just kinda curious at this point.

Reply:
I was also curious, and it was easy enough to expand this test to see! It does indeed block reads: 7fee2649d